projects/whats_cooking_good_looking/whats_cooking_good_looking/train_ner_workflow.py

```diff
@@ -0,0 +1,261 @@
+import json
+import pickle
+import random
+from collections import defaultdict
+from pathlib import Path
+
+import spacy
+from flytekit import Resources, dynamic, task, workflow
+from spacy.language import Language
+from spacy.training import Example
+from spacy.util import compounding, minibatch
+
+from whats_cooking_good_looking.utils import (download_bytes_from_gcs,
+                                              download_from_gcs, load_config,
+                                              upload_to_gcs)
+
+SPACY_MODEL = {"en": "en_core_web_sm"}
+
+CACHE_VERSION = "2.2"
+request_resources = Resources(cpu="1", mem="500Mi", storage="500Mi")
+limit_resources = Resources(cpu="2", mem="1000Mi", storage="1000Mi")
+
+THRESHOLD_ACCURACY = 0.7
+
+
+@task
+def evaluate_ner(labelstudio_tasks: bytes) -> dict:
+    """Computes accuracy, precision and recall of NER model out of label studio output.
+
+    Args:
+        labelstudio_tasks (list): List of dicts outputs of label studio annotation with following format
+                        [
+                            {
+                            "result": [
+                                {
+                                    "value": {"start": 10, "end": 17, "text": "Chennai", "labels": ["LOC"]},
+                                    "from_name": "label",
+                                    "to_name": "text",
+                                    "type": "labels",
+                                    "origin": "manual",
+                                }
+                            ],
+                            "predictions": [
+                                    {
+                                    "result": {"start": 10, "end": 17, "text": "Chennai", "labels": ["LOC"]},
+                                    "model_version": "dummy",
```

Comment:
What is model_version? Could you elaborate?

Reply:
I think it's "en_core_web_sm", correct? If so, would we have other model versions apart from this?